src/index.js

```diff
@@ -38,42 +37,53 @@ export class Konfettikanone extends Component {
 
   generateConfetti() {
     for (let i = 0; i < 100; i++) {
-      const randomValues = this.calcRandomValues();
-      const type = `down${randomValues.type}`;
+      const {size, left, color: backgroundColor, speed, type: typeValue} = this.calcRandomValues();
+      const {current: target} = this.confettiWrapper;
+      const type = `down${typeValue}`;
 
       const inlineStyles = {
-        height: randomValues.size,
-        width: randomValues.size,
-        left: randomValues.left,
-        backgroundColor: randomValues.color,
+        height: size,
+        width: size,
+        left,
+        backgroundColor,
         animation: `${styles[type](
-          `${this.confettiWrapper.current.clientHeight + 20}px`
-        )} ${randomValues.speed}s linear ${Math.random() * 3}s 1 forwards`
+          `${target.clientHeight + 20}px`
+        )} ${speed}s linear ${Math.random() * 3}s 1 forwards`
       };
 
       const particle = document.createElement("div");
       Object.assign(particle.style, inlineStyles);
       particle.className = styles.particle;
 
-      this.confettiWrapper.current.appendChild(particle);
+      target.appendChild(particle);
     }
   }
 
   calcRandomValues() {
-    const size = `${Math.floor(Math.random() * 3) + 7}px`;
-    const color = this.props.colors[
-      Math.floor(Math.random() * this.props.colors.length)
-    ];
-    const type = this.props.type[
-      Math.floor(Math.random() * this.props.type.length)
-    ];
-    const speed = this.props.speed[
-      Math.floor(Math.random() * this.props.speed.length)
-    ];
-    const left = `${Math.floor(
-      Math.random() * this.confettiWrapper.current.clientWidth
-    )}px`;
-
-    return {size, left, color, type, speed};
+    const {colors, types, speed: speeds} = this.props
+    const {current: target} = this.confettiWrapper;
+
+    const size = `${Konfettikanone.random(3) + 7}px`;
+    const color = Konfettikanone.pickRandomElement(colors);
+    const type = Konfettikanone.pickRandomElement(types);
+    const speed = Konfettikanone.pickRandomElement(speeds);
+    const left = `${Konfettikanone.random(target.clientWidth)}px`;
+
+    return {
+      size,
+      left,
+      color,
+      type,
+      speed
+    };
+  }
+
+  static pickRandomElement(a) {
+    return a[Konfettikanone.random(a.length)]
   }
+
+  static random(n = 1) {
+    return Math.floor(Math.random() * n);
```

Comment:
Can we use `Math.round()` with `a.length - 1` instead?
If we use `Math.floor()` and `Math.random()` evaluates to `1` we will get an out of range exception.

Reply:
Could we also add tests for this?

Reply:
> Can we use `Math.round()` with `a.length - 1` instead?
> If we use `Math.floor()` and `Math.random()` evaluates to `1` we will get an out of range exception.

Nice catch!